### PR TITLE
Remove Explicit Deletion of CMake `/GR` Flag as 3.20's *CMP0117* by Default Doesn't Add It

### LINKED ISSUE #94354: Enable CMake policy CMP0117 and update the corresponding compile options throughout the repo.

### PR DESCRIPTION
**Pull request by ivdiazsa**
Completes #94354. CMake Version 3.20+ implements a new version of Policy CMP0117. This new version does not add the `/GR` flag by default to `CMAKE_CXX_FLAGS`. And this is the behavior we want. However, previous versions of CMake did add it, so we had to manually remove it. This PR removes that deletion as part of the efforts to simplify our CMake build in general.

### CODE REVIEW

File: src/coreclr/CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,8 @@ cmake_minimum_required(VERSION 3.20)
 # Set the project name
 project(CoreCLR)
 
+cmake_policy(SET CMP0117 NEW)
```
[the remark this answers is not in the document]
Comment (ivdiazsa) [reply src/coreclr/CMakeLists.txt]: The Ninja files were the same, so indeed not specifying = set by default. I've sent a new commit with the change removing it.

File: eng/native/configurecompiler.cmake [hunk 2]
```diff
@@ -783,11 +783,6 @@ if (MSVC)
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/GS>) # Explicitly enable the buffer security checks
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/fp:precise>) # Enable precise floating point
 
-  # disable C++ RTTI
-  # /GR is added by default by CMake, so remove it manually.
-  string(REPLACE "/GR " " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR-")
```
Comment (jkoritzinsky) [inline eng/native/configurecompiler.cmake:789]: We still want to add `/GR-` as the default is equivalent to specifying `/GR` (which is what prompted the CMake policy).

**Review by jkoritzinsky — CHANGES_REQUESTED** [review]

File: eng/native/configurecompiler.cmake [hunk 3]
```diff
@@ -783,9 +783,6 @@ if (MSVC)
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/GS>) # Explicitly enable the buffer security checks
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/fp:precise>) # Enable precise floating point
 
-  # disable C++ RTTI
```
Comment (jkoritzinsky) [inline eng/native/configurecompiler.cmake:786]: Let's keep this comment

**Review by jkoritzinsky — APPROVED** [review]
LGTM with one nit

**Comment by ivdiazsa** [comment]
Failure is unrelated and is being tracked in issue #97103, so we can merge this now.

**Pull request merged**